esphome/components/fan/__init__.py

```diff
@@ -143,6 +150,9 @@ async def fan_turn_off_to_code(config, action_id, template_arg, args):
             cv.Required(CONF_ID): cv.use_id(FanState),
             cv.Optional(CONF_OSCILLATING): cv.templatable(cv.boolean),
             cv.Optional(CONF_SPEED): cv.templatable(cv.int_range(1)),
+            cv.Optional(CONF_DIRECTION, default="FORWARD"): cv.templatable(
```

Comment:
This really should not have a default if it is supposed to be optional. With a default of forward, it can have a side effect of changing the direction if the user does not specify direction when changing the speed while it is in reverse mode.

```suggestion
            cv.Optional(CONF_DIRECTION): cv.templatable(
```

Reply:
Good catch, thanks!